internal/identity/identity.go

```diff
@@ -0,0 +1,199 @@
+package identity
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/types/optional"
+	"google.golang.org/protobuf/proto"
+
+	v1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type Identity interface {
+	String() string
+}
+
+var _ Identity = Controller{}
+
+type Controller struct{}
+
+func NewController() Controller {
+	return Controller{}
+}
+
+func (c Controller) String() string {
+	return "ca"
+}
+
+var _ Identity = Runner{}
+
+// Runner identity
+// TODO: Maybe use KeyType[T any, TP keyPayloadConstraint[T]]?
+type Runner struct {
+	Key    model.RunnerKey
+	Module string
+}
+
+func NewRunner(key model.RunnerKey, module string) Runner {
+	return Runner{
+		Key:    key,
```

Comment:
This is the FTL "key", as in the unique workload identifier at from a platform perspective. I think this constitutes of the host, port and a nonce.